packages/flutter_tools/lib/src/dart/pub.dart

```diff
@@ -519,12 +547,20 @@ class _DefaultPub implements Pub {
       homeDirectory = _platform.environment['Home'] ?? '';
     }
 
-    if (_fileSystem.directory(cachePath).existsSync()) {
+   final Directory globalDirectory = _fileSystem.directory(_fileSystem.path.join(homeDirectory, '.pub-cache'));
+
+    if (needsToJoinCache(
+        fileSystem: _fileSystem,
```

Comment:
style nit: de-dent once and add a trailing comma to line 555

Reply:
Here, and lines 564, 567, 571-573, please de-dent once and add trailing commas where there aren't one already